components/dashboard/src/app/AppLoading.tsx

```diff
@@ -5,8 +5,19 @@
  */
 
 import { FunctionComponent } from "react";
+import { Delayed } from "../components/Delayed";
+import { Heading3, Subheading } from "../components/typography/headings";
+import gitpodIcon from "../icons/gitpod.svg";
 
-// TODO: Would be great to show a loading UI if it's been loading for awhile
 export const AppLoading: FunctionComponent = () => {
-    return <></>;
+    return (
+        // Wait 3 seconds before showing the loading screen to avoid flashing it too quickly
+        <Delayed wait={3000}>
+            <div className="flex flex-col justify-center items-center w-full h-screen space-y-4">
+                <img src={gitpodIcon} alt="Gitpod's logo" className={"h-16 flex-shrink-0 animate-bounce"} />
```

Comment:
**thought**: Re-using the bouncing logo could frustrate or confuse users as they usually to see that loading state (bouncing logo) a lot while opening workspaces. Could be worth using something different, as the designs included in https://github.com/gitpod-io/gitpod/issues/16803. Thoughts?

Reply:
Ah good point. I could also remove the animation, definitely don't want user's to get confused. I mentioned it below, but I think the small little loading spinner is quite easy to not notice by itself. Would love some personality injected into the product here too 😄

Reply:
DEAL—Opened #17044 to keep track of the proposal to improve the full-page loading.